src/test/java/org/jabref/gui/exporter/SaveDatabaseActionTest.java

```diff
@@ -91,6 +103,48 @@ public void saveShouldShowSaveAsIfDatabaseNotSelected() {
         verify(saveDatabaseAction, times(1)).saveAs(file);
     }
 
+    @Test
+    public void saveShouldMarkAllWrittenBibEntriesAsUnchanged() throws Exception {
+        file = Files.createTempFile("JabRef", ".bib");
+
+        BibEntry firstEntry = new BibEntry().withField(StandardField.AUTHOR, "first");
+        firstEntry.setChanged(true);
+        BibEntry secondEntry = new BibEntry().withField(StandardField.AUTHOR, "second");
+        secondEntry.setChanged(true);
+        BibDatabase database = new BibDatabase(List.of(firstEntry, secondEntry));
+
+        LatexFieldFormatterPreferences latexFieldFormatterPreferences = mock(LatexFieldFormatterPreferences.class);
+        when(latexFieldFormatterPreferences.getFieldContentParserPreferences()).thenReturn(mock(FieldContentParserPreferences.class));
+        SavePreferences savePreferences = mock(SavePreferences.class);
+        dbContext = mock(BibDatabaseContext.class);
```

Comment:
There shouldn't be any need to reinit these mocks here (if they are already initialized in the setup method)

Reply:
One cannot overwrite `thenReturn` unless the mock is recreated. (Tried it multiple times)